packages/core/saved-objects/core-saved-objects-server-internal/src/saved_objects_service.ts

```diff
@@ -147,9 +147,13 @@ export class SavedObjectsService
 
     registerCoreObjectTypes(this.typeRegistry);
 
+    const skipMigration = this.config.migration.skip;
+
     return {
       status$: calculateStatus$(
-        this.migrator$.pipe(switchMap((migrator) => migrator.getStatus$())),
+        skipMigration
+          ? of({ status: 'completed' })
```

Comment:
Could have introduced a new `skipped` status, but it was additional work and thinking about it I wasn't sure to see any upside, so I just used `completed` instead.

Tell me if you think it would be preferable to have a dedicated migration status for this scenario.